lib/mosaic-database/src/main/java/org/eclipse/mosaic/lib/database/road/Connection.java

```diff
@@ -265,9 +265,20 @@ public void applyTurnRestriction(@Nonnull Restriction.Type type, @Nonnull Connec
                 // this means there should be ONLY the target in the connection
                 outgoing.clear();
                 outgoing.put(target.getId(), target);
+
+                for (Connection outgoingConnection : getTo().getOutgoingConnections()) {
+                    if (!outgoingConnection.getId().equals(target.getId())) {
+                        outgoingConnection.incoming.remove(getId());
+                    }
+                }
                 break;
             case Not:
                 outgoing.remove(target.getId());
+                for (Connection outgoingConnection : getTo().getOutgoingConnections()) {
```

Comment:
why is the loop necessary in that case? for the `Not` case we explicitly define which incoming->outgoing relation is prohibited. Therefore I would assume a simple `outgoing.incoming.remove(getId())` is sufficient. 

assume connection `A` is an incoming connection of node `X` and connections `B` and `C` are outgoing connection of node `X`. If you only want to prohibit `A->X->B` but still allow `A->X->C`, you should only remove A from B`.incoming`, not from `C.incoming`

Reply:
Yes that's true. I was quite confused by this since `incoming` is a private field and `getIncoming()` returns an `unmodifiable`  collection. So I was confused why we could access it at all. However it seemed to work in the `for`-loop so I just went with it. What also seems to work is casting `outgoing` to a `Connection` beforehand: `((Connection) outgoing).incoming.remove(getId())`.

Maybe you can shed some light on this 😅

Reply:
Yes, the scope of the member `incoming` is private. The private scope is not limited on objects, but on the class only. Since you are inside the class `Connection` you also have access on private members of any object of class `Connection`. No need to cast whatsoever.

Reply:
Ah I just spotted my thinking error. It should be `target.incoming.remove(getId())` not `outgoing.incoming.remove(getId())`. 🤦‍♂️

Reply:
ups